lang/cs/Org.Apache.REEF.IO/DataCache.cs

```diff
@@ -0,0 +1,434 @@
+            {
+                CleanHigherLevelCache(CacheLevel.InMemoryAsStream);
+            }
+
+            _memStreams = memStreams;
+            _lowestCacheLevel = CacheLevel.InMemoryAsStream;
+            return _lowestCacheLevel;
+        }
+
+        /// <summary>
+        /// "Materializes" and deserializes the data. Also caches to memory.
+        /// To not cache to memory, please use the <see cref="Materialize"/> method.
+        /// </summary>
+        /// <returns>The deserialized data.</returns>
+        public IEnumerable<T> MaterializeAndCache(bool shouldCleanHigherLevelCache)
+        {
+            return Materialize(true, shouldCleanHigherLevelCache);
+        }
+
+        /// <summary>
+        /// "Materializes" and deserializes the data. Does not cache.
+        /// To cache, please use the <see cref="MaterializeAndCache"/> method.
+        /// </summary>
+        /// <returns>The deserialized data.</returns>
+        public IEnumerable<T> Materialize()
+        {
+            return Materialize(false, false);
+        }
+
+        private IEnumerable<T> Materialize(bool shouldCache, bool shouldCleanHigherLevelCache)
+        {
+            if (shouldCache)
+            {
+                CacheToMaterialized(shouldCleanHigherLevelCache);
+            }
+
+            switch (_lowestCacheLevel)
+            {
+                case CacheLevel.InMemoryAsStream:
+                    CheckMemoryStreams(true);
+                    return _inputDataMover.MemoryToMaterialized(_memStreams.Value);
+                case CacheLevel.Disk:
+                    CheckDisk(true);
+                    return _inputDataMover.DiskToMaterialized(_diskDirectory.Value);
+                case CacheLevel.NotLocal:
+                    return _inputDataMover.RemoteToMaterialized();
+                case CacheLevel.InMemoryMaterialized:
+                    CheckMaterialized(true);
+                    return _materialized.Value;
+                default:
+                    throw new IllegalStateException("Illegal cache layer.");
+            }
+        }
+
+        /// <summary>
+        /// Deserialize and cache the data in memory from the level the data is currently at.
+        /// </summary>
+        private CacheLevel CacheToMaterialized(bool shouldCleanHigherLevelCache)
+        {
+            Optional<IReadOnlyCollection<T>> materialized;
+
+            switch (_lowestCacheLevel)
+            {
+                case CacheLevel.InMemoryAsStream:
+                    CheckMemoryStreams(true);
+                    materialized = Optional<IReadOnlyCollection<T>>.Of(
+                        new List<T>(_inputDataMover.MemoryToMaterialized(_memStreams.Value)));
+                    break;
+                case CacheLevel.Disk:
+                    CheckDisk(true);
+                    materialized = Optional<IReadOnlyCollection<T>>.Of(
+                        new List<T>(_inputDataMover.DiskToMaterialized(_diskDirectory.Value)));
```

Comment:
As per discussion offline, I think confusion was the comment to the function. Consider changing it to: 
"Deserialize and cache the materialized data in memory from the level the data is currently at ."